### PR TITLE
Add 'use-submgr' toggle-option and functionality

### LINKED ISSUE #49: [BUG] Prevent Annoying Entitlement Server Errors in EC2s
**Describe the bug**

After creating AMIs, resultant EC2s complain:

~~~
This system is not registered with an entitlement server. You can use subscription-manager to register.
~~~

Each time the yum service is used, prompting complaints from EC2-users

**Severity**

- [ ] Completely Broken (No work-around evident)
- [ ] Severely Broken (Work-around possible but difficult)
- [ ] Moderately Broken (Trivial work-around)
- [X] Nuisance (Functions but untrapped errors can slip through)

**To Reproduce**
1. Build AMI
2. Launch EC2 from AMI
3. Attempt to perform a `yum update`
4. `yum update` activities occur, but are preceded by the previously-quoted error message

**Expected behavior**

Yum works with no annoying entitlement server error messages

**Fix Suggestions**

Ensure that `enabled` parameter is set to `0` in the newly-created AMI's `/etc/yum/pluginconf.d/subscription-manager.conf`

### PR DESCRIPTION
**Pull request by ferricoxide**
Set default behavior to disabling attempts to use the Subscription-Manager service. Allow it to be specifically enabled by selecting the command toggle-option.

Closes #49

### CODE REVIEW

**Comment by ferricoxide** [comment]
Created test-AMI using `spel-commercial` job `3ec615e7-39f2-49b8-9b54-27c449bc0c34` to build (private) image `ami-0442483d6ed324ce6`.

**Comment by lorengordon** [comment]
Is this something that may be disabled on non-RHEL, but that must be enabled on RHEL? Or another way, is subscription manager required for RHEL, the way they're built "license-included"?

**Comment by ferricoxide** [comment]
> Is this something that may be disabled on non-RHEL, but that must be enabled on RHEL? Or another way, is subscription manager required for RHEL, the way they're built "license-included"?

No. It only needs to be `enabled` on systems that _actually_ use it. Obviously, CentOS/OLx/etc. don't actually use it, but neither do RHUI-enabled systems. It's specifically for systems that manage RPM-access through either RHN or Satellite.

File: PostBuild.sh [hunk 1]
```diff
@@ -480,12 +482,28 @@ function authselectInit {
...
+function DisableSubscriptionManager {
+  local YUM_CONF
+  YUM_CONF=$( readlink -f /etc/yum/pluginconf.d/subscription-manager.conf )
+  if [[ ${SUBSCRIPTION_MANAGER} == "enabled" ]]
+  then
+     return
+  fi
+
+  err_exit "Attempting to disable subscription-manager service... " NONE
+  chroot "${CHROOTMNT}" /bin/sed -i '/^enabled/s/1$/0/' "${YUM_CONF}" || \
```
Comment (lorengordon) [inline PostBuild.sh:495]: Do we need to gate this on existence of whatever package installs this file, or maybe just on the existence of the file?

Comment (ferricoxide) [reply PostBuild.sh:495]: Yeah. For some reason I thought it was an RPM with a dependency-graph. However:

```bash
# rpm -q --whatrequires $( rpm --qf '%{name}\n' -qf $( readlink -f /etc/yum/pluginconf.d/subscription-manager.conf ) )
no package requires subscription-manager
# rpm -q --whatprovides $( rpm --qf '%{name}\n' -qf $( readlink -f /etc/yum/pluginconf.d/subscription-manager.conf ) )
subscription-manager-1.28.36-3.el8_8.x86_64
```

So, I'll add a file-existence check.

Comment (ferricoxide) [reply PostBuild.sh:495]: Added the existence-check to the function's initial-exit criteria.

File: PostBuild.sh [hunk 2]
```diff
@@ -480,12 +482,28 @@ function authselectInit {
    err_exit "Succeeded initializing authselect" NONE
 }
 
+# Disable subscription-manager
+function DisableSubscriptionManager {
+  local YUM_CONF
+  YUM_CONF=$( readlink -f /etc/yum/pluginconf.d/subscription-manager.conf )
```
Comment (lorengordon) [inline PostBuild.sh:488]: Won't the readlink will return non-zero if not found, causing `set -e` will abort?

Comment (ferricoxide) [reply PostBuild.sh:488]: Ok, so, `subscription-manager` RPM is part of the `@core` RPM-group. So, it _should_ always be present (given the intended build-scenario. That said, since we notionally support using a manifest-file in place of relying on repositories containing group-metadata, probably need to test for the `subscription-manager` RPM as a hedge against that edge use-case. So, I'll re-order the function's content, some.

**Review by lorengordon — APPROVED** [review]
